source/core/sdirk_stage_data.cc

```diff
@@ -0,0 +1,35 @@
+#include <core/sdirk_stage_data.h>
+
+#include <deal.II/base/logstream.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <iomanip>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+using namespace dealii;
+
+SDIRKStageData
+sdirk_stage_data(const FullMatrix<double>  &butcher_table,
+                 const std::vector<double> &c,
+                 const std::vector<double> &b,
+                 const unsigned int         stage_i)
+{
+  const unsigned int n_stages = b.size();
+
+  if (stage_i >= n_stages)
+    throw std::invalid_argument("Invalid stage index.");
+
+  SDIRKStageData data;
+  data.a_ij.resize(stage_i + 1);
```

Comment:
I think since you are resizing it from the outside, it could be a better idea to have SDIRKStageData be a class and the constructor could take asa n argument the size to be resized.
Here  it's not clear why it's stage_i+1. Be explicit and add a comment here.